universe/auto_syncer.go

```diff
@@ -219,55 +233,173 @@ func (f *FederationEnvoy) syncServerState(ctx context.Context,
 	return nil
 }
 
-// pushProofToFederation attempts to push out a new proof to the current
-// federation in parallel.
-func (f *FederationEnvoy) pushProofToFederation(uniID Identifier, key LeafKey,
-	leaf *Leaf) {
+// pushProofToServer attempts to push out a new proof to the target server.
+func (f *FederationEnvoy) pushProofToServer(ctx context.Context,
+	uniID Identifier, key LeafKey, leaf *Leaf, addr ServerAddr) error {
 
-	// Fetch all universe servers in our federation.
-	fedServers, err := f.tryFetchServers()
-	if err != nil || len(fedServers) == 0 {
-		return
+	remoteUniverseServer, err := f.cfg.NewRemoteRegistrar(addr)
+	if err != nil {
+		return fmt.Errorf("cannot push proof unable to connect "+
+			"to remote server(%v): %w", addr.HostStr(), err)
 	}
 
-	log.Infof("Pushing new proof to %v federation members, proof_key=%v",
-		len(fedServers), spew.Sdump(key))
+	_, err = remoteUniverseServer.UpsertProofLeaf(
+		ctx, uniID, key, leaf,
+	)
+	if err != nil {
+		return fmt.Errorf("cannot push proof to remote "+
+			"server(%v): %w", addr.HostStr(), err)
+	}
 
-	ctx, cancel := f.WithCtxQuitNoTimeout()
-	defer cancel()
+	return nil
+}
+
+// pushProofToServerLogged attempts to push out a new proof to the target
+// server, and logs the sync attempt.
+func (f *FederationEnvoy) pushProofToServerLogged(ctx context.Context,
```

Comment:
nit: can use `pushProofToServer` in here

Reply:
Good point, thanks.